client/state/happychat/test/middleware.js

```diff
@@ -92,4 +99,60 @@ describe( 'middleware', () => {
 			expect( connection.request ).toHaveBeenCalledWith( action, action.timeout );
 		} );
 	} );
+
+	describe( 'eventMessage', () => {
```

Comment:
These new tests are great! To be 100% sure that it won't regress again in the future, could we also add the cases for `HAPPYCHAT_FOCUS` and all the other non-connected state? Namely `HAPPYCHAT_CONNECTION_STATUS_CONNECTING`, `HAPPYCHAT_CONNECTION_STATUS_DISCONNECTED`,`HAPPYCHAT_CONNECTION_STATUS_RECONNECTING` and ``HAPPYCHAT_CONNECTION_STATUS_UNAUTHORIZED`.
It's basically the same test code with ( 2 actions ) x ( 5 status ) data variation. From the php world, we can achieve that by data provider. There is also [jasmine-data-provider](https://www.npmjs.com/package/jasmine-data-provider) for Jasmine test framework. I'm not sure if there is anything like this from Jest. If not, I think we can just elaborate the combinations in an array and iterate through in the test.

Reply:
fixed at f72c4d3afeb2830b319cf5a2b6aab88c750efc69